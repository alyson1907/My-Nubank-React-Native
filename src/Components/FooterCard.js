import React from 'react'
import Text from './Text'
import { Image } from 'react-native'
import { useNavigation } from '@react-navigation/native'
import { TouchableOpacity } from 'react-native-gesture-handler'

import Icon from 'react-native-vector-icons/AntDesign'
import Colors from '../Colors'
import styles from './Styles/FooterCard'

const FooterCard = (props) => {
  const navigation = useNavigation()
  const navigateTo = pageName => pageName && navigation.navigate(pageName)

  return (
    // TODO add link to navigate to another screen
    <TouchableOpacity style={styles.container} onPress={() => navigateTo(props.navigateTo)}>
      <Icon name={props.icon} size={22} color={Colors.secondary} />
      <Text style={styles.contentText}>{props.text}</Text>
    </TouchableOpacity>
  )
}

export default FooterCard
