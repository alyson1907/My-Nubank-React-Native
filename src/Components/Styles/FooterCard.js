import { StyleSheet } from 'react-native'
import Colors from '../../Colors'

const styles = StyleSheet.create({
  container: {
    width: 120,
    height: 120,
    padding: 14,
    marginHorizontal: 8,
    backgroundColor: Colors.primaryAlternative,
    borderRadius: 8,
    justifyContent: 'space-between'
  },

  contentText: {
    color: Colors.fontPrimary,
    fontWeight: 'bold',
    fontSize: 16
  }
})

export default styles
