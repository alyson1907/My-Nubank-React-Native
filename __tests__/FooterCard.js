import React from 'react'
import 'react-native'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'
import FooterCard from '../src/Components/FooterCard'

const globals = {
  text: 'random test text',
  icon: 'creditcard'
}

describe('Footer Card Tests', () => {
  it('evaluates the rendered card with snapshot', () => {
    const tree = renderer.create(<FooterCard  icon={globals.icon} text={globals.text} />)
    expect(tree).toMatchSnapshot()
  })

  it('renders single card correctly', () => {
    // Mounting Component
    const wrapper = mount(<FooterCard icon={globals.icon} text={globals.text} />)
    const footerBtn = wrapper.find("[testID='footer-card']").first()
    const footerTxt = wrapper.find("[testID='footer-card-text']").first().text()

    expect(footerBtn).toBeTruthy()
    expect(footerTxt).toEqual(globals.text)
  })
})
