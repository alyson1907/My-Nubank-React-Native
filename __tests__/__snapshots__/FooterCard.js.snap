// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen Footer List Tests evaluates the rendered card with snapshot 1`] = `
<RNGestureHandlerButton
  collapsable={false}
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
  style={
    Array [
      Object {
        "overflow": "hidden",
      },
      undefined,
    ]
  }
>
  <View
    accessible={true}
    style={
      Object {
        "backgroundColor": "#ac14d9",
        "borderRadius": 8,
        "height": 120,
        "justifyContent": "space-between",
        "marginHorizontal": 8,
        "opacity": 1,
        "padding": 18,
        "width": 120,
      }
    }
    testID="footer-card"
  >
    <Icon
      color="#eee"
      name="creditcard"
      size={22}
    />
    <Text
      style={
        Array [
          Object {
            "fontFamily": "sans-serif-medium",
          },
          Object {
            "color": "#eee",
            "fontSize": 16,
            "fontWeight": "bold",
          },
        ]
      }
      testID="footer-card-text"
    >
      random test text
    </Text>
  </View>
</RNGestureHandlerButton>
`;
